Sources/StreamChatUI/CommonViews/Attachments/AttachmentPreviewProvider.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+public protocol AttachmentPreviewProvider {
+    /// The view representing the attachment.
+    func previewView<ExtraData: ExtraDataTypes>(components: _Components<ExtraData>) -> UIView
+    
+    /// The preferred axis to be used for attachment previews in attachments view.
+    static var preferredAxis: NSLayoutConstraint.Axis { get }
+}
+
+public extension AttachmentPreviewProvider {
+    static var preferredAxis: NSLayoutConstraint.Axis { .horizontal }
+}
```

Comment:
Why do we need a default implementation here? It's easier for who is implementing this protocol to decide the axis.

Reply:
fixed, thanks!